Code/Framework/AzFramework/AzFramework/Physics/RigidBodyBus.h

```diff
@@ -79,13 +81,68 @@ namespace Physics
 
     using RigidBodyRequestBus = AZ::EBus<RigidBodyRequests>;
 
-
+    //! Notifications interface for a rigid body (static or dynamic).
     class RigidBodyNotifications
         : public AZ::ComponentBus
     {
+    private:
+        template<class Bus>
+        struct RigidBodyNotificationsConnectionPolicy : public AZ::EBusConnectionPolicy<Bus>
+        {
+            static void Connect(
+                typename Bus::BusPtr& busPtr,
+                typename Bus::Context& context,
+                typename Bus::HandlerNode& handler,
+                typename Bus::Context::ConnectLockGuard& connectLock,
+                const typename Bus::BusIdType& id = 0)
+            {
+                AZ::EBusConnectionPolicy<Bus>::Connect(busPtr, context, handler, connectLock, id);
+
+                AZ::Entity* entity = nullptr;
+                EBUS_EVENT_RESULT(entity, AZ::ComponentApplicationBus, FindEntity, id);
+                if (entity)
+                {
+                    // Only immediately dispatch if the entity is already active, otherwise when
+                    // entity will get activated it will make the notifications itself.
+                    const AZ::Entity::State entityState = entity->GetState();
+                    if (entityState == AZ::Entity::State::Active)
+                    {
+                        // Only immediately dispatch if the entity is a RigidBodyRequestBus' handler.
+                        RigidBodyRequestBus::EnumerateHandlersId(
+                            id,
+                            [handler, id](RigidBodyRequests* rigidBodyhandler)
+                            {
+                                if (rigidBodyhandler->IsPhysicsEnabled())
+                                {
+                                    handler->OnPhysicsEnabled(id);
+                                }
+                                else
+                                {
+                                    handler->OnPhysicsDisabled(id);
+                                }
+                                return true;
+                            });
+                    }
+                }
+            }
+        };
+
     public:
-        virtual void OnPhysicsEnabled() = 0;
-        virtual void OnPhysicsDisabled() = 0;
+        /**
```

Comment:
I think we're almost entirely using the //! style in AzFramework/Physics

Reply:
Comment was copy/paste from other places in the code, I can change it.